<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="eDITH">
<title>An introduction to eDITH • eDITH</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="An introduction to eDITH">
<meta property="og:description" content="eDITH">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">eDITH</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.2.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/eDITH.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/lucarraro/eDITH/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>An introduction to eDITH</h1>
                        <h4 data-toc-skip class="author">Luca Carraro,
Florian Altermatt</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/lucarraro/eDITH/blob/HEAD/vignettes/eDITH.Rmd" class="external-link"><code>vignettes/eDITH.Rmd</code></a></small>
      <div class="d-none name"><code>eDITH.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p><code>eDITH</code> (<strong>eD</strong>NA
<strong>I</strong>ntegrating <strong>T</strong>ransport and
<strong>H</strong>ydrology) allows spatially explicit interpretation and
prediction of environmental DNA (eDNA) data collected from river
networks. It implements the eDITH model, which couples a
geomorphological and hydrological characterization of a catchment, eDNA
transport and decay dynamics, and a species distribution model, to
transform pointwise water eDNA data (sensu <span class="citation">(Pawlowski, Apothéloz-Perret-Gentil, and Altermatt
2020)</span>) collected at a catchment into space-filling predicted maps
of taxon density.</p>
<p>Features:</p>
<ul>
<li>It provides estimations of detection probability of any number of
taxa across a whole catchment based on spatially replicated eDNA data of
these taxa.</li>
<li>It can handle both DNA concentration data (e.g., from qPCR) and
metabarcoding (read counts) data of any taxonomic group, and can be used
to predict space-filling maps of density of taxa (say, at species, genus
or family level), OTUs (Operational Taxonomic Units) or ASVs (Amplicon
Sequence Variants).</li>
<li>Model fit can be performed via Bayesian techniques or optimization
algorithms.</li>
<li>Covariates can be specified by the user and/or selected by means of
Asymmetric Eigenvector Maps (AEMs).</li>
<li>An interface to the <code>DHARMa</code> package for residual
diagnostics is provided.</li>
</ul>
<p><code>eDITH</code> requires the use of river networks defined as
<code>river</code> objects, which can be built via the
<code>rivnet</code> package.</p>
</div>
<div class="section level2">
<h2 id="installing-the-package">Installing the package<a class="anchor" aria-label="anchor" href="#installing-the-package"></a>
</h2>
<p><code>eDITH</code> can be installed from CRAN:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"eDITH"</span><span class="op">)</span></span></code></pre></div>
<p>The development version can be installed from Github:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"lucarraro/eDITH"</span><span class="op">)</span></span></code></pre></div>
<p>Among other packages, <code>eDITH</code> depends on
<code>rivnet</code>, which in turn depends on the <code>traudem</code>
package. <code>traudem</code> relies on the <a href="https://hydrology.usu.edu/taudem/taudem5/" class="external-link">TauDEM</a> library.
<code>traudem</code> provides a guide to correct installation of TauDEM
and its dependencies for different operating systems, and offers wrapper
commands to call TauDEM methods from R. Please read the
<code>traudem</code> <a href="https://lucarraro.github.io/traudem/" class="external-link">documentation</a>
carefully.</p>
</div>
<div class="section level2">
<h2 id="the-edith-model-in-a-nutshell">The eDITH model in a nutshell<a class="anchor" aria-label="anchor" href="#the-edith-model-in-a-nutshell"></a>
</h2>
<div class="section level3">
<h3 id="overview-and-underlying-assumptions">Overview and underlying assumptions<a class="anchor" aria-label="anchor" href="#overview-and-underlying-assumptions"></a>
</h3>
<p>The eDITH model <span class="citation">(Carraro et al. 2018; Carraro,
Mächler, et al. 2020; Carraro, Stauffer, and Altermatt 2021; Carraro,
Blackman, and Altermatt 2023)</span> makes use of spatially replicated
eDNA measurements within a river network to infer the spatial
distribution of any taxon (or OTU, or ASV; hereafter simply “taxon”) of
interest across that river network. The key underlying concept is that
eDNA particles are advected downstream by streamflow, and hence an eDNA
sample is not only representative of the location where it is taken, but
it provides information about taxon abundance (in the case of qPCR
single-species data) or biodiversity (for metabarcoding data) for a
certain area upstream of the sampling location. By exploiting
information from multiple sampling sites distributed in space, and
embedding a model for transport and concurrent decay of DNA, the eDITH
model is able to disentangle the various sources of DNA shedding (and
hence, of the target taxon’s abundance). As a result, eDITH complements
pointwise eDNA measurements by projecting taxon distributions (and thus
biodiversity information) into space-filling catchment maps.</p>
<div class="section level4">
<h4 id="dna-production-and-decay-rates">DNA production and decay rates<a class="anchor" aria-label="anchor" href="#dna-production-and-decay-rates"></a>
</h4>
<p>The DNA production (i.e., shedding) rate of a taxon in stream water
is assumed to be proportional to its density<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content='&lt;p&gt;In particular, it is assumed that taxon density is
equivalent to both biomass and abundance. Although shedding rates have
been shown to scale allometrically with organismal mass &lt;span class="citation"&gt;(Yates et al. 2020)&lt;/span&gt;, the distinction between
biomass and abundance is, for modelling purposes, irrelevant. In
essence, the patterns modelled by eDITH are those of a quantity that is
proportional to the DNA production rate–be it biomass, abundance or
allometrically scaled mass.&lt;/p&gt;'><sup>1</sup></a> <span class="citation">(Lodge et al. 2012; Apothéloz-Perret-Gentil et al.
2017)</span>. The validity of this assumption is restricted to the time
point when eDNA is sampled. Thus, it is irrelevant if a taxon varies in
its release of DNA across the season, as long as DNA released is
proportionate to taxon density at the time of sampling. For the same
reason, hydrological variables (such as discharge and velocity) are
assumed to be constant in time (not in space!) across the duration of a
sampling campaign. For instance, if multiple days are required to
perform eDNA sampling across all sampling sites, these days would be
assumed to constitute a unique time point, during which hydrological
conditions can be adequately represented by time-averaged values of
discharge and velocity.</p>
<p>The decay of DNA is assumed to be expressed by first order kinetics
(i.e., the rate of change of DNA concentration decreases linearly with
time). Similarly to the production rate, it is assumed that a single
value of decay time is representative for a given taxon at a given time
point.</p>
</div>
<div class="section level4">
<h4 id="sampling-design">Sampling design<a class="anchor" aria-label="anchor" href="#sampling-design"></a>
</h4>
<p>The spatial extent of the eDNA sampling must adequately cover the
entire river network, in order to allow the model to make robust
predictions. As a rule of thumb, there should be one sampling site per
each 10–20 km<sup>2</sup> of drained area.</p>
<p>Moreover, all main tributaries of a river network should be sampled
in a spatially hierarchical design. Ideally, sampling sites should be
located just upstream of a confluence, so that the independent signals
from the joining tributaries can be gauged. See <span class="citation">(Carraro, Stauffer, and Altermatt 2021)</span> for
further information on optimal eDNA sampling strategies.</p>
<p>The mean distance (along the river network) between eDNA samples
should not be too large, otherwise the eDNA found at an upstream site
would be totally depleted before reaching the next downstream site. In
this case, the eDITH model could not make inferences based on these
multiple sites. To make an example, if we consider first order kinetics,
a water velocity of 1 ms<sup>-1</sup> and a decay time of 4 h (i.e., a
half life of 2.77 h), we would be able to measure only about 50% of the
concentration that we would measure 10 km upstream, and 3% of the
concentration that we would measure 50 km upstream<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;This estimate considers a channel with no lateral
inputs. If lateral inputs with water not containing the target eDNA were
considered, these percentages would inevitably decrease&lt;/p&gt;"><sup>2</sup></a>. As a rule of thumb,
sites that are more than 50 km apart (along the river network) should be
considered as independent samples. See also <span class="citation">(Deiner and Altermatt 2014; Pont et al. 2018)</span>
with respect to transport distances of eDNA.</p>
<p>Finally, the model assumes that eDNA is well mixed in the water
column, and/or that samples at a site adequately span the river cross
section (e.g., sampling at both banks and at the centre, for large
rivers, while in small rivers sampling at a single bank might be
sufficient, by assuming that eDNA is sufficiently well mixed). Analogous
care should be taken in the choice of a representative sampling volume
to be filtered <span class="citation">(Altermatt et al.
2023)</span>.</p>
</div>
<div class="section level4">
<h4 id="measurement-errors">Measurement errors<a class="anchor" aria-label="anchor" href="#measurement-errors"></a>
</h4>
<p>Variability in DNA detection through sampling (e.g., filtration) or
laboratory procedures (e.g. DNA extraction, Polymerase Chain Reaction
(PCR), or sequencing) do not introduce a systematic bias in DNA
concentrations across samples.</p>
<p>If eDITH is used to model metabarcoding data, the expected read count
at a given site for a given taxon is assumed to be proportional to the
underlying DNA concentration.</p>
<p>Replicated eDNA measures at the same site and time point are treated
as independent measures. No distinctions is made on whether these are
physical (i.e., different water samples) or laboratory (different PCR
runs) replicates.</p>
</div>
<div class="section level4">
<h4 id="what-taxa-can-be-modelled">What taxa can be modelled?<a class="anchor" aria-label="anchor" href="#what-taxa-can-be-modelled"></a>
</h4>
<p>In principle, the eDITH model is suitable for aquatic macroorganisms
that shed their DNA into stream water. Nonetheless, the model can also
be extended for microorganisms that are sampled in their entirety (e.g.,
free floating bacteria), in which case their upstream sources can be
considered as biofilm colonies, and the decay time does not refer to DNA
molecules but rather to the bacteria lifetime.</p>
<p>The eDITH model does not make a distinction on whether the eDNA data
is referred to an assigned taxon (say, at a species, genus or family
level) or an unassigned cluster such as OTUs or ASVs. What matters is
that the assumptions on production and decay rates (see <a href="#dna-production-and-decay-rates">DNA production and decay
rates</a>) must hold.</p>
<p>In principle, it is also possible to estimate the spatial
distribution of terrestrial, rather than aquatic, taxa, provided that
the assumption of proportionality between DNA production rate and taxon
density holds. In this case, the spatial unit on which predictions are
performed is the subcatchment, i.e., the portion of land that directly
drains towards the associated reach.</p>
</div>
<div class="section level4">
<h4 id="river-network-model">River network model<a class="anchor" aria-label="anchor" href="#river-network-model"></a>
</h4>
<p>The domain on which the eDITH model is applied is a river network
discretized into reaches, i.e. segments of river not interrupted by
confluences and treated as smallest spatial units. Each reach is
considered as a node of the network, and the ensemble of the reaches
covers the entire river network. It is assumed that reaches have no
internal variability (e.g., the exact coordinates of a sampling site do
not matter, provided that the sampling site is associated to the same
reach). Note that the along-stream distance between two consecutive
confluences can be covered by a sequence of different reaches, to allow
for a finer discretization of the river network. Indeed, number and
maximum length of reaches can be tuned by using function
<code>aggregate_river</code> of <code>rivnet</code>. See also <span class="citation">(Carraro, Bertuzzo, et al. 2020)</span> and the
documentation of the <code>OCNet</code> package for details on
aggregation of a river network into reaches.</p>
</div>
<div class="section level4">
<h4 id="lakes-and-braided-channels">Lakes and braided channels<a class="anchor" aria-label="anchor" href="#lakes-and-braided-channels"></a>
</h4>
<p>The eDITH model is designed to work for river networks not containing
lakes or reservoirs. Dynamics of eDNA in lakes are still largely
unclear, but the very large residence times of water particles in lakes
compared to those of a river reach imply that all dissolved DNA entering
a lake either degrades or is retained by substrate particles.
Particle-bound DNA can eventually be resuspended <span class="citation">(Shogren et al. 2017)</span>, but this likely occurs at
long time scales (say, monthly, yearly or even longer), hence the
assumption of first-order decay kinetics with a single decay time value
across the river network would be violated. Similarly, the river
networks produced by <code>rivnet</code> do not admit bifurcations in
the downstream direction (e.g., braided or artificial channels creating
loops in the river network). Hence, the different braids or channels
cannot be treated as independent reaches in the eDITH model. In such a
case, the eDITH model could still be applied by considering a single
channel as a conceptual equivalent of the multiple real braids or
channels.</p>
</div>
</div>
<div class="section level3">
<h3 id="the-governing-equations">The governing equations<a class="anchor" aria-label="anchor" href="#the-governing-equations"></a>
</h3>
<p>The main equation of the eDITH model results from a mass balance of
eDNA across a cross-section of a river, and it reads:</p>
<p><span class="math display">\[\begin{equation}
C_j = \frac{1}{Q_j}\sum_{i \in \gamma(j)} p_i A_{S,i}
\exp\left(-\frac{L_{ij}}{\overline{v_{ij}} \tau}\right)
\end{equation}\]</span></p>
<p>where:</p>
<ul>
<li>
<span class="math inline">\(C_j\)</span> is the DNA concentration at
a sampling site <span class="math inline">\(j\)</span> (strictly
speaking, <span class="math inline">\(j\)</span> is the reach where a
sampling site is located);</li>
<li>
<span class="math inline">\(Q_j\)</span> is the water discharge in
<span class="math inline">\(j\)</span>;</li>
<li>
<span class="math inline">\(\gamma(j)\)</span> identifies the set of
reaches upstream of <span class="math inline">\(j\)</span>;</li>
<li>
<span class="math inline">\(p_i\)</span> is the DNA production rate
at an upstream reach <span class="math inline">\(i\)</span>;</li>
<li>
<span class="math inline">\(A_{S,i}\)</span> is the source area in
<span class="math inline">\(i\)</span> (i.e., extent of the node;
considering a river reach and an aquatic taxon, this could e.g. be equal
to the product of its length and width);</li>
<li>
<span class="math inline">\(L_{ij}\)</span> is the length of the
along-stream path joining <span class="math inline">\(i\)</span> to
<span class="math inline">\(j\)</span>;</li>
<li>
<span class="math inline">\(\overline{v_{ij}}\)</span> is the
average water velocity along <span class="math inline">\(L_{ij}\)</span>;</li>
<li>
<span class="math inline">\(\tau\)</span> is a characteristic decay
time for DNA in stream water.</li>
</ul>
<p>Assuming that the morphology and hydrology of the river network are
known (and thus lengths, areas and discharges), the above equation links
DNA concentrations to the unknown parameters <span class="math inline">\(\tau\)</span> and <span class="math inline">\(\mathbf{p}=\left( p_1, \dots, p_N\right)\)</span>
(where <span class="math inline">\(N\)</span> is the total number of
reaches). While the former is linked to the behaviour of DNA in stream
water (and could in principle be measured, or at least its value be
constrained), the estimation of the latter is the actual goal of the
eDITH model. Contrasting observed and modelled DNA concentrations thus
enables the estimation of maps of <span class="math inline">\(\mathbf{p}\)</span> across all <span class="math inline">\(N\)</span> reaches constituting the river network,
and hence of relative taxon density, given the initial assumption.</p>
<p>It is often convenient (both from a modelling and interpretation
viewpoint) to express the DNA production rate <span class="math inline">\(p_i\)</span> as a function of environmental
covariates, possibly related to the spatial patterns of the investigated
taxon:</p>
<p><span class="math display">\[\begin{equation}
p_i = p_0 \exp\left( \boldsymbol\beta^T \mathbf{X}(i) \right)
\end{equation}\]</span></p>
<p>where <span class="math inline">\(\mathbf{X}(i)\)</span> is a vector
of covariates evaluated at reach <span class="math inline">\(i\)</span>;
<span class="math inline">\(\boldsymbol\beta\)</span> a vector of
covariate effect sizes; and <span class="math inline">\(p_0\)</span> a
baseline production rate, i.e., the DNA production rate at a site where
all covariates are at a null level. This reduces the number of unknowns
from <span class="math inline">\(N\)</span> (size of <span class="math inline">\(\mathbf{p}\)</span>) to the number of selected
covariates (length of <span class="math inline">\(\boldsymbol\beta\)</span>) plus one (<span class="math inline">\(p_0\)</span>).</p>
<p>If the eDNA data are in the form of read counts, the above equations
remain applicable. Indeed, <span class="math inline">\(C_j\)</span>
would play the role of the expected read number at site <span class="math inline">\(j\)</span> (proportional to the underlying DNA
concentration, as per the initial hypothesis). Consequently, <span class="math inline">\(\mathbf{p}\)</span> would in this case represent
the DNA production rates multiplied by such constant of
proportionality.</p>
</div>
<div class="section level3">
<h3 id="estimating-detection-probability">Estimating detection probability<a class="anchor" aria-label="anchor" href="#estimating-detection-probability"></a>
</h3>
<p>Once model parameters are estimated (either via a Bayesian method or
likelihood maximization), it is possible to transform DNA production
rates <span class="math inline">\(\mathbf{p}\)</span> into corresponding
detection probabilities. This is done by exploiting the assumption on
the probability distribution used to model measurement errors (and hence
formulate the likelihood, see <a href="#likelihood-function-and-model-parameters">Likelihood function and
model parameters</a>).</p>
<p>In particular, this is done by calculating, for each reach <span class="math inline">\(j\)</span>, the expected eDNA value <span class="math inline">\(\widetilde{C_j}\)</span> (concentration or read
count) if that reach were detached from the river network (that is, in
the absence of upstream inputs, with the water discharge in the reach
being equal to the locally produced discharge<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content='&lt;p&gt;Calculated as the actual discharge &lt;span class="math inline"&gt;\(Q_j\)&lt;/span&gt; minus the sum of discharges of the
upstream reaches that are directly connected to &lt;span class="math inline"&gt;\(j\)&lt;/span&gt;&lt;/p&gt;'><sup>3</sup></a>). This is then
transformed into a detection probability value, calculated as the
probability of observing a non-null eDNA value under the assumed error
probability distribution and the expected value <span class="math inline">\(\widetilde{C_j}\)</span>.</p>
<p>The so-obtained detection probability maps can be further transformed
into presence-absence maps and used to assess biodiversity patterns
<span class="citation">(e.g., Carraro, Blackman, and Altermatt
2023)</span>.</p>
</div>
</div>
<div class="section level2">
<h2 id="required-data">Required data<a class="anchor" aria-label="anchor" href="#required-data"></a>
</h2>
<p>eDNA data must be provided in the <code>data</code> field of
<code>run_eDITH_BT</code> and <code>run_eDITH_optim</code> as a data
frame with components <code>values</code> (eDNA values measured for a
given taxon and a given time point) and <code>ID</code> (identifiers of
the network nodes (i.e., reaches) at the AG level<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;Please refer to the documentation of &lt;code&gt;OCNet&lt;/code&gt;
for an overview of the aggregation levels of a &lt;code&gt;river&lt;/code&gt;
object.&lt;/p&gt;"><sup>4</sup></a> where the sampling
sites are located). To identify the latter, function
<code>locate_site</code> from <code>rivnet</code> can be used.</p>
<p>The river network must be provided as a <code>river</code> object,
obtained via the <code>rivnet</code> package. It is fundamental that the
river be aggregated into reaches (via <code>aggregate_river</code>) and
that it contains hydrological data, such that discharges and water
velocities can be used as input in the eDITH model. This can be obtained
via the <code>hydro_river</code> function of <code>rivnet</code>. In its
simplest setting, a single value of discharge (or depth) and width, not
necessarily at the same reach, are required in order to extrapolate
hydraulic variables across the whole network. The exact method with
which hydrological variables are extrapolated to the river network
depends on the number and type of data provided; please refer to the
documentation of <code>hydro_river</code> for further details.</p>
<p>Optionally, covariates can be passed to <code>run_eDITH_BT</code> and
<code>run_eDITH_optim</code> as a data frame. Function
<code>covariate_river</code> of <code>rivnet</code> allows computing
covariate values from raster maps and a <code>river</code> object. If
covariates are not provided, asymmetric eigenvector maps (AEMs) are
calculated on the river network and used as covariates. AEMs <span class="citation">(Blanchet, Legendre, and Borcard 2008)</span> are
mutually orthogonal spatial variables obtained by a spatial filtering
technique that considers space in an asymmetric way, and are thus
suitable to model species distributions in river networks. It is of
course possible to combine user-provided covariates and AEMs, as
detailed in the flowchart below:</p>
<div class="figure">
<img src="flowchart_cov.png" alt="Flowchart for the choice of covariates used to fit the eDITH model." width="90%"><p class="caption">
Flowchart for the choice of covariates used to fit the eDITH model.
</p>
</div>
</div>
<div class="section level2">
<h2 id="likelihood-function-and-model-parameters">Likelihood function and model parameters<a class="anchor" aria-label="anchor" href="#likelihood-function-and-model-parameters"></a>
</h2>
<p>By default, <code>eDITH</code> implements four probability
distributions to model errors between observed and modelled data:</p>
<ul>
<li>the normal (<code>ll.type = "norm"</code>) and log-normal
(<code>ll.type = "lnorm"</code>) distributions are suitable to model
<em>DNA concentration</em> data.</li>
<li>the geometric (<code>ll.type = "geom"</code>) and negative binomial
(<code>ll.type = "nbinom"</code>) distributions are suitable for
<em>read count</em> data.</li>
</ul>
<p><strong>Note that the user is expected to set option
<code>ll.type</code> according to the nature of the <code>data</code>
provided in <code>run_eDITH_BT</code> or
<code>run_eDITH_optim</code>.</strong></p>
<p>Moreover, the optional argument <code>no.det = TRUE</code> can be
used to produce a zero-inflated error distribution. This is mandatory if
<code>ll.type = "lnorm"</code>, as the log-normal distribution does not
admit zeros.</p>
<p>Default model parameters are as follows:</p>
<ul>
<li>
<code>tau</code> represents decay time <span class="math inline">\(\tau\)</span> in hours.</li>
<li>
<code>log_p0</code> is the logarithm in base 10 of the baseline
production rate <span class="math inline">\(p_0\)</span>. The units of
<span class="math inline">\(p_0\)</span> are equal to the units of the
inputted <code>data$values</code> multiplied by a length unit and
divided by a time unit. For instance, if <code>data$values</code>
contains DNA concentration data in mol m<sup>-3</sup> and the river
object contains discharge (<code>river$AG$discharge</code>) in
m<sup>3</sup>s<sup>-1</sup> and areas<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content='&lt;p&gt;This depends on the resolution of the DEM used to obtain
the river network in &lt;code&gt;&lt;a href="https://lucarraro.github.io/rivnet/reference/extract_river.html" class="external-link"&gt;rivnet::extract_river&lt;/a&gt;&lt;/code&gt;.&lt;/p&gt;'><sup>5</sup></a> in m<sup>2</sup>, <span class="math inline">\(p_0\)</span> is expressed in mol m<sup>-2</sup>
s<sup>-1</sup> (i.e. amount of DNA shed per unit area and unit time). In
the case of read count data, the unit for <span class="math inline">\(p_0\)</span> might look weird; this is because, in
this case, <span class="math inline">\(p_0\)</span> embeds a constant
that transforms expected DNA concentration predicted by the model into
expected read number.</li>
<li>
<code>beta_X</code> is the effect size of covariate <code>X</code>,
where <code>X</code> is replaced by the object names in the
<code>covariates</code> data frame (if provided), and/or by
e.g. <code>AEM1</code> when AEM eigenfunctions are used. In this case,
the number indicates the corresponding eigenfunction.</li>
</ul>
<p>Additional parameters can be added depending on the likelihood
definition:</p>
<ul>
<li>
<code>sigma</code> is the standard deviation of the error (when
<code>ll.type = "norm"</code> or <code>"lnorm"</code>). Note that, if a
log-normal distribution is used, <code>sigma</code> is the standard
deviation of the non-log-transformed values (and is hence different than
parameter <code>sdlog</code> of <code>dlnorm</code>).</li>
<li>
<code>omega</code> is the overdispersion parameter (when
<code>ll.type = "nbinom"</code>), which is defined as the ratio between
variance and mean <span class="citation">(Lindén and Mäntyniemi
2011)</span>.</li>
<li>no additional error parameter is present if
<code>ll.type = "geom"</code>.</li>
<li>
<code>Cstar</code> is a further parameter added when the optional
argument <code>no.det = TRUE</code> is passed. In this case, the
probability of non-detection is expressed as <code>exp(-C/Cstar)</code>,
where <code>C</code> is the modelled eDNA value (expected concentration
or read number) at a given site. Increased <code>Cstar</code> increases
the non-detection probability, all else being equal. See <span class="citation">Carraro et al. (2018)</span> for details.</li>
</ul>
</div>
<div class="section level2">
<h2 id="a-minimal-example">A minimal example<a class="anchor" aria-label="anchor" href="#a-minimal-example"></a>
</h2>
<div class="section level3">
<h3 id="generating-a-river-object">Generating a <code>river</code> object<a class="anchor" aria-label="anchor" href="#generating-a-river-object"></a>
</h3>
<p>To illustrate the functioning of <code>eDITH</code>, let’s run an
example in the river Wigger (Switzerland), whose <code>river</code>
object can be called via <code>data(wigger)</code>. Such an object can
be built via the <code>rivnet</code> package as follows. Please refer to
<code>rivnet</code> documentation for details on its functions.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Extract river from DEM</span></span>
<span><span class="va">river</span> <span class="op">&lt;-</span> <span class="fu">rivnet</span><span class="fu">::</span><span class="fu"><a href="https://lucarraro.github.io/rivnet/reference/extract_river.html" class="external-link">extract_river</a></span><span class="op">(</span>outlet<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">637478</span>,<span class="fl">237413</span><span class="op">)</span>,</span>
<span>                    EPSG<span class="op">=</span><span class="fl">21781</span>, <span class="co">#CH1903/LV03 coordinate system</span></span>
<span>                    ext<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">6.2e5</span>,<span class="fl">6.6e5</span>,<span class="fl">2e5</span>,<span class="fl">2.5e5</span><span class="op">)</span>,</span>
<span>                    z<span class="op">=</span><span class="fl">9</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Aggregate river - default thrA and maxReachLength = 2500 m</span></span>
<span><span class="va">river</span> <span class="op">&lt;-</span> <span class="fu">rivnet</span><span class="fu">::</span><span class="fu"><a href="https://lucarraro.github.io/rivnet/reference/aggregate_river.html" class="external-link">aggregate_river</a></span><span class="op">(</span><span class="va">river</span>, maxReachLength <span class="op">=</span> <span class="fl">2500</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Hydraulic data: width = 8 m, discharge = 15 m3/s at the outlet</span></span>
<span><span class="va">hydrodata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">8</span>, <span class="fl">15</span><span class="op">)</span>, </span>
<span>                        type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"w"</span>, <span class="st">"Q"</span><span class="op">)</span>, </span>
<span>                        node <span class="op">=</span> <span class="va">river</span><span class="op">$</span><span class="va">AG</span><span class="op">$</span><span class="va">outlet</span><span class="op">*</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Assign hydraulic variables across the river network</span></span>
<span><span class="va">river</span> <span class="op">&lt;-</span> <span class="fu">rivnet</span><span class="fu">::</span><span class="fu"><a href="https://lucarraro.github.io/rivnet/reference/hydro_river.html" class="external-link">hydro_river</a></span><span class="op">(</span><span class="va">hydrodata</span>, <span class="va">river</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Attribute landcover classes as covariates</span></span>
<span><span class="va">r1</span> <span class="op">&lt;-</span> <span class="fu">terra</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/terra/man/rast.html" class="external-link">rast</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/landcover.tif"</span>, </span>
<span>                       package <span class="op">=</span> <span class="st">"rivnet"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">river</span> <span class="op">&lt;-</span> <span class="fu">rivnet</span><span class="fu">::</span><span class="fu"><a href="https://lucarraro.github.io/rivnet/reference/covariate_river.html" class="external-link">covariate_river</a></span><span class="op">(</span><span class="va">r1</span>, <span class="va">river</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="preparing-and-displaying-edna-data">Preparing and displaying eDNA data<a class="anchor" aria-label="anchor" href="#preparing-and-displaying-edna-data"></a>
</h3>
<p>An example of DNA concentration data for a given target taxon can be
called as:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">dataC</span><span class="op">)</span></span></code></pre></div>
<p><code>dataC</code> contains the following objects:</p>
<ul>
<li>
<code>ID</code>: identifiers of river network nodes at the AG levels
where eDNA has been sampled (these can be retrieved via
<code><a href="https://lucarraro.github.io/rivnet/reference/locate_site.html" class="external-link">rivnet::locate_site</a></code> based on the site coordinates).</li>
<li>
<code>values</code>: corresponding values of DNA concentration
measured (in mol m<sup>-3</sup>)</li>
</ul>
<p><code>dataC</code> contains eDNA values for 24 sampling sites, each
of which has been sampled three times. In such case, the three different
eDNA values are inserted in <code>dataC</code> as independent rows:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dataC</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">dataC</span><span class="op">$</span><span class="va">ID</span><span class="op">==</span><span class="fl">2</span><span class="op">)</span>,<span class="op">]</span></span>
<span><span class="co">#&gt;    ID       values</span></span>
<span><span class="co">#&gt; 1   2 0.000000e+00</span></span>
<span><span class="co">#&gt; 25  2 1.037331e-12</span></span>
<span><span class="co">#&gt; 49  2 8.176798e-13</span></span></code></pre></div>
<p>We can use function <code><a href="https://lucarraro.github.io/rivnet/reference/points_colorscale.html" class="external-link">rivnet::points_colorscale</a></code> to display
the mean DNA concentration values across the sampling sites:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sites</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">dataC</span><span class="op">$</span><span class="va">ID</span><span class="op">)</span></span>
<span><span class="va">values</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">sites</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">ind</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">sites</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">s</span> <span class="op">&lt;-</span> <span class="va">sites</span><span class="op">[</span><span class="va">ind</span><span class="op">]</span></span>
<span>  <span class="va">values</span><span class="op">[</span><span class="va">ind</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">dataC</span><span class="op">$</span><span class="va">values</span><span class="op">[</span><span class="va">dataC</span><span class="op">$</span><span class="va">ID</span><span class="op">==</span><span class="va">s</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">river</span><span class="op">)</span></span>
<span><span class="fu">rivnet</span><span class="fu">::</span><span class="fu"><a href="https://lucarraro.github.io/rivnet/reference/points_colorscale.html" class="external-link">points_colorscale</a></span><span class="op">(</span><span class="va">river</span><span class="op">$</span><span class="va">AG</span><span class="op">$</span><span class="va">X</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">dataC</span><span class="op">$</span><span class="va">ID</span><span class="op">)</span><span class="op">]</span>, <span class="va">river</span><span class="op">$</span><span class="va">AG</span><span class="op">$</span><span class="va">Y</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">dataC</span><span class="op">$</span><span class="va">ID</span><span class="op">)</span><span class="op">]</span>,</span>
<span>                          <span class="va">values</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/title.html" class="external-link">title</a></span><span class="op">(</span><span class="st">"Mean observed DNA concentration [mol m-3]"</span><span class="op">)</span></span></code></pre></div>
<p><img src="map2.png" width="80%"></p>
</div>
<div class="section level3">
<h3 id="running-the-edith-model">Running the eDITH model<a class="anchor" aria-label="anchor" href="#running-the-edith-model"></a>
</h3>
<p>We can define a set of covariates containing main landcover classes
(urban areas, agricultural areas and forest) in addition to elevation
and log-scaled drainage area:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">covariates</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>urban <span class="op">=</span> <span class="va">river</span><span class="op">$</span><span class="va">SC</span><span class="op">$</span><span class="va">locCov</span><span class="op">$</span><span class="va">landcover_1</span>,</span>
<span>                         agriculture <span class="op">=</span> <span class="va">river</span><span class="op">$</span><span class="va">SC</span><span class="op">$</span><span class="va">locCov</span><span class="op">$</span><span class="va">landcover_2</span>,</span>
<span>                         forest <span class="op">=</span> <span class="va">river</span><span class="op">$</span><span class="va">SC</span><span class="op">$</span><span class="va">locCov</span><span class="op">$</span><span class="va">landcover_3</span>,</span>
<span>                         elev <span class="op">=</span> <span class="va">river</span><span class="op">$</span><span class="va">AG</span><span class="op">$</span><span class="va">Z</span>,</span>
<span>                         log_drainageArea <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="va">river</span><span class="op">$</span><span class="va">AG</span><span class="op">$</span><span class="va">A</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="section level4">
<h4 id="bayesian-methods">Bayesian methods<a class="anchor" aria-label="anchor" href="#bayesian-methods"></a>
</h4>
<p>We can use a Bayesian approach to fit the eDITH model. Function
<code>run_eDITH_BT</code> interfaces with the <code>BayesianTools</code>
package, which contains several different Bayesian samplers. The default
choice in <code>run_eDITH_BT</code> is DREAMzs <span class="citation">(Carraro, Blackman, and Altermatt 2023)</span>. We can
run eDITH with the above-specified covariates:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">out.bt.cov</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/run_eDITH_BT.html">run_eDITH_BT</a></span><span class="op">(</span><span class="va">dataC</span>, <span class="va">river</span>, <span class="va">covariates</span><span class="op">)</span></span></code></pre></div>
<p>The default option for <code>ll.type</code> is <code>norm</code>,
which is suitable for DNA concentration data.</p>
<p>Alternatively, we can run eDITH via AEM covariates. By default, the
first 10% AEM eigenfunctions are used as covariates.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">out.bt.aem</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/run_eDITH_BT.html">run_eDITH_BT</a></span><span class="op">(</span><span class="va">dataC</span>, <span class="va">river</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="optimization">Optimization<a class="anchor" aria-label="anchor" href="#optimization"></a>
</h4>
<p>Alternatively, an optimization method can be used to estimate the
best-fit parameters of the eDITH model. Function
<code>run_eDITH_optim</code> internally calls <code>optim</code> and its
default Nelder-Mead method. The main options such as
<code>covariates</code> and <code>ll.type</code> work in the same way as
in <code>run_eDITH_BT</code>:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">27</span><span class="op">)</span></span>
<span><span class="va">out.opt.aem</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/run_eDITH_optim.html">run_eDITH_optim</a></span><span class="op">(</span><span class="va">dataC</span>, <span class="va">river</span>, n.AEM <span class="op">=</span> <span class="fl">10</span>, </span>
<span>    n.attempts <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> </span></code></pre></div>
<p>For illustrative purposes, we here use a reduced number of AEM
eigenfunctions (10) and a single optimization attempt. It is advisable
to increase the number of optimization attempts (as in the default
settings) to ensure convergence.</p>
<p>We can qualitatively check the goodness of fit of the model:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">out.opt.aem</span><span class="op">$</span><span class="va">C</span><span class="op">[</span><span class="va">dataC</span><span class="op">$</span><span class="va">ID</span><span class="op">]</span>, <span class="va">dataC</span><span class="op">$</span><span class="va">values</span>, </span>
<span>     xlim<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">8e-12</span><span class="op">)</span>, ylim<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">8e-12</span><span class="op">)</span>, asp<span class="op">=</span><span class="fl">1</span>,</span>
<span>     xlab <span class="op">=</span> <span class="st">"Modelled concentration [mol m-3]"</span>, </span>
<span>     ylab <span class="op">=</span> <span class="st">"Observed concentration [mol m-3]"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/abline.html" class="external-link">abline</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="Cobs.png" width="80%"></p>
<p>And also display modelled concentrations across the river network, by
using <code>rivnet</code>’s <code>plot</code> method:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">river</span>, <span class="va">out.opt.aem</span><span class="op">$</span><span class="va">C</span>, colLevels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">values</span><span class="op">)</span>, <span class="fl">1000</span><span class="op">)</span>, addLegend <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>     colPalette <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/palettes.html" class="external-link">hcl.colors</a></span><span class="op">(</span><span class="fl">1000</span>, <span class="st">"Reds 3"</span>, rev<span class="op">=</span><span class="cn">T</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">rivnet</span><span class="fu">::</span><span class="fu"><a href="https://lucarraro.github.io/rivnet/reference/points_colorscale.html" class="external-link">points_colorscale</a></span><span class="op">(</span><span class="va">river</span><span class="op">$</span><span class="va">AG</span><span class="op">$</span><span class="va">X</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">dataC</span><span class="op">$</span><span class="va">ID</span><span class="op">)</span><span class="op">]</span>, <span class="va">river</span><span class="op">$</span><span class="va">AG</span><span class="op">$</span><span class="va">Y</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">dataC</span><span class="op">$</span><span class="va">ID</span><span class="op">)</span><span class="op">]</span>,</span>
<span>                          <span class="va">values</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/title.html" class="external-link">title</a></span><span class="op">(</span><span class="st">"DNA concentration [mol m-3]"</span><span class="op">)</span></span></code></pre></div>
<p><img src="mapC.png" width="80%"></p>
<p>Finally, we can have a look at the predicted map of relative taxon
density (i.e., DNA production rate <span class="math inline">\(\mathbf{p}\)</span>)</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">river</span>, <span class="va">out.opt.aem</span><span class="op">$</span><span class="va">p</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/title.html" class="external-link">title</a></span><span class="op">(</span><span class="st">'DNA production rate [mol m-2 s-1]'</span><span class="op">)</span></span></code></pre></div>
<p><img src="mapp.png" width="80%"></p>
<p>And the corresponding detection probability <span class="math inline">\(\mathbf{p_D}\)</span>:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">river</span>, <span class="va">out.opt.aem</span><span class="op">$</span><span class="va">probDet</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/title.html" class="external-link">title</a></span><span class="op">(</span><span class="st">'Detection probability'</span><span class="op">)</span></span></code></pre></div>
<p><img src="mappD.png" width="80%"></p>
</div>
</div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-altermatt2023" class="csl-entry">
Altermatt, F., L. Carraro, M. Antonetti, C. Albouy, Y. Zhang, A. Lyet,
X. Zhang, and L. Pellissier. 2023. <span>“Quantifying Biodiversity Using
eDNA from Water Bodies. General Principles and Recommendations for
Sampling Designs.”</span> <em>Environmental DNA</em>. <a href="https://doi.org/10.1002/edn3.430" class="external-link">https://doi.org/10.1002/edn3.430</a>.
</div>
<div id="ref-apotheloz2017" class="csl-entry">
Apothéloz-Perret-Gentil, L., A. Cordonier, F. Straub, J. Iseli, and J.
Pawlowski. 2017. <span>“Taxonomy-Free Molecular Diatom Index for
High-Throughput eDNA Biomonitoring.”</span> <em>Molecular Ecology
Resources</em>. <a href="https://doi.org/10.1111/1755-0998.12668" class="external-link">https://doi.org/10.1111/1755-0998.12668</a>.
</div>
<div id="ref-blanchet2008" class="csl-entry">
Blanchet, F. G., P. Legendre, and D. Borcard. 2008. <span>“Modelling
Directional Spatial Processes in Ecological Data.”</span> <em>Ecological
Modelling</em>. <a href="https://doi.org/10.1016/j.ecolmodel.2008.04.001" class="external-link">https://doi.org/10.1016/j.ecolmodel.2008.04.001</a>.
</div>
<div id="ref-Carraro2020OCN" class="csl-entry">
Carraro, L., E. Bertuzzo, E. A. Fronhofer, R. Furrer, I. Gounand, A.
Rinaldo, and F. Altermatt. 2020. <span>“Generation and Application of
River Network Analogues for Use in Ecology and Evolution.”</span>
<em>Ecology and Evolution</em>. <a href="https://doi.org/10.1002/ece3.6479" class="external-link">https://doi.org/10.1002/ece3.6479</a>.
</div>
<div id="ref-Carraro2023" class="csl-entry">
Carraro, L., R. C. Blackman, and F. Altermatt. 2023. <span>“Modelling
Environmental DNA Transport in Rivers Reveals Highly Resolved
Spatio-Temporal Biodiversity Patterns.”</span> <em>Scientific
Reports</em>. <a href="https://doi.org/10.1038/s41598-023-35614-6" class="external-link">https://doi.org/10.1038/s41598-023-35614-6</a>.
</div>
<div id="ref-carraro2018" class="csl-entry">
Carraro, L., H. Hartikainen, J. Jokela, E. Bertuzzo, and A. Rinaldo.
2018. <span>“Estimating Species Distribution and Abundance in River
Networks Using Environmental DNA.”</span> <em>Proceedings of the
National Academy of Sciences of the United States of America</em>. <a href="https://doi.org/10.1073/pnas.1813843115" class="external-link">https://doi.org/10.1073/pnas.1813843115</a>.
</div>
<div id="ref-Carraro2020" class="csl-entry">
Carraro, L., E. Mächler, R. Wüthrich, and F. Altermatt. 2020.
<span>“Environmental DNA Allows Upscaling Spatial Patterns of
Biodiversity in Freshwater Ecosystems.”</span> <em>Nature
Communications</em>. <a href="https://doi.org/10.1038/s41467-020-17337-8" class="external-link">https://doi.org/10.1038/s41467-020-17337-8</a>.
</div>
<div id="ref-Carraro2021" class="csl-entry">
Carraro, L., J. B. Stauffer, and F. Altermatt. 2021. <span>“How to
Design Optimal eDNA Sampling Strategies for Biomonitoring in River
Networks.”</span> <em>Environmental DNA</em>. <a href="https://doi.org/10.1002/edn3.137" class="external-link">https://doi.org/10.1002/edn3.137</a>.
</div>
<div id="ref-Deiner2014" class="csl-entry">
Deiner, K., and F. Altermatt. 2014. <span>“Transport Distance of
Invertebrate Environmental DNA in a Natural River.”</span> <em>PLOS
ONE</em>. <a href="https://doi.org/10.1371/journal.pone.0088786" class="external-link">https://doi.org/10.1371/journal.pone.0088786</a>.
</div>
<div id="ref-linden2011" class="csl-entry">
Lindén, A., and S. Mäntyniemi. 2011. <span>“Using the Negative Binomial
Distribution to Model Overdispersion in Ecological Count Data.”</span>
<em>Ecology</em>. <a href="https://doi.org/10.1890/10-1831.1" class="external-link">https://doi.org/10.1890/10-1831.1</a>.
</div>
<div id="ref-lodge2012" class="csl-entry">
Lodge, D. M., C. R. Turner, C. L. Jerde, M. A. Barnes, L. Chadderton, S.
P. Egan, J. L. Feder, A. R. Mahon, and M. E. Pfrender. 2012.
<span>“Conservation in a Cup of Water. Estimating Biodiversity and
Population Abundance from Environmental DNA.”</span> <em>Molecular
Ecology</em>. <a href="https://doi.org/10.1111/j.1365-294X.2012.05600.x" class="external-link">https://doi.org/10.1111/j.1365-294X.2012.05600.x</a>.
</div>
<div id="ref-pawlowski2020" class="csl-entry">
Pawlowski, J., L. Apothéloz-Perret-Gentil, and F. Altermatt. 2020.
<span>“Environmental DNA. What’s Behind the Term? Clarifying the
Terminology and Recommendations for Its Future Use in
Biomonitoring.”</span> <em>Molecular Ecology</em>. <a href="https://doi.org/10.1111/mec.15643" class="external-link">https://doi.org/10.1111/mec.15643</a>.
</div>
<div id="ref-Pont2018" class="csl-entry">
Pont, D., M. Rocle, A. Valentini, R. Civade, J. Pauline, A. Maire, N.
Roset, M. Schabuss, H. Zornig, and T. Dejean. 2018. <span>“Environmental
DNA Reveals Quantitative Patterns of Fish Biodiversity in Large Rivers
Despite Its Downstream Transportation.”</span> <em>Scientific
Reports</em>. <a href="https://doi.org/10.1038/s41598-018-28424-8" class="external-link">https://doi.org/10.1038/s41598-018-28424-8</a>.
</div>
<div id="ref-Shogren2017" class="csl-entry">
Shogren, A. J., J. L. Tank, E. Andruszkiewicz, B. Olds, A. R. Mahon, C.
L. Jerde, and D. Bolster. 2017. <span>“Controls on eDNA Movement in
Streams. Transport, Retention, and Resuspension.”</span> <em>Scientific
Reports</em>. <a href="https://doi.org/10.1038/s41598-017-05223-1" class="external-link">https://doi.org/10.1038/s41598-017-05223-1</a>.
</div>
<div id="ref-yates2020" class="csl-entry">
Yates, M. C., D. M. Glaser, J. R. Post, M. E. Cristescu, D. J. Fraser,
and A. M. Derry. 2020. <span>“The Relationship Between eDNA Particle
Concentration and Organism Abundance in Nature Is Strengthened by
Allometric Scaling.”</span> <em>Molecular Ecology</em>. <a href="https://doi.org/10.1111/mec.15543" class="external-link">https://doi.org/10.1111/mec.15543</a>.
</div>
</div>
</div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Luca Carraro, Florian Altermatt, University of Zurich.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
